### PR TITLE
Fixed issue where callback_metrics was replaced instead of updated

### LINKED ISSUE #490: Early stopping conditioned on metric `val_loss` isn't recognised when setting the val_check_interval
**Describe the bug**
Training stops when setting `val_check_interval`<1.0 in the Trainer class as it doesn't recognise `val_loss`. I get the following warning at the end of the 3rd epoch:
```
Early stopping conditioned on metric `val_loss` which is not available. Available metrics are: loss,train_loss
```

**To Reproduce**
1. Run the CoolModel example but change the `trainer` line to
trainer = Trainer(val_check_interval=0.5,default_save_path="test")    
2. Training will stop at the end of the third epoch and the above warning will show.

**Expected behavior**
Training shouldn't stop and `val_loss` should be recognised.

**Desktop (please complete the following information):**
 - VM: Google Colab
 - Version 0.5.3.2

**Additional context**
This doesn't happen with 0.5.2.1 although it looks like something has changed with model saving mechanism since it only seems to save the best model in 0.5.3.2.

EDIT: Also seems to happen when setting `train_percent_check`<1.0

### PR DESCRIPTION
**Pull request by ryanwongsa**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/williamFalcon/pytorch-lightning/blob/master/.github/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  

## What does this PR do?
Fixes #490 .

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by williamFalcon** [comment]
@ryanwongsa  thanks!
